airflow/models/taskinstance.py

```diff
@@ -722,6 +722,32 @@ def error(self, session=NEW_SESSION):
         session.merge(self)
         session.commit()
 
+    @classmethod
+    @provide_session
+    def find(
+        cls,
+        task_id: str,
+        dag_id: str,
+        execution_date: Optional[datetime] = None,
+        run_id: Optional[str] = None,
+        session: Session = NEW_SESSION,
+    ) -> bool:
+        """Returns a TaskInstance for the given criteria or none."""
```

Comment:
```suggestion
        """Find a TaskInstance for the given criteria.

        Paramters ``dag_id`` and ``task_id`` are required. Exactly one of ``run_id`` and
        ``execution_date`` is required to find the DAG run that owns the task instance.
        """
```

But can we _not_ support `execution_date` here? A long-term goal of AIP-39 is to remove `DagRun.execution_date`’s uniqueness requirement (so two runs can have the same logical date as long as they have different run IDs), and I’m hesitant to add new APIs that rely on this unique constraint.